storage/local/interface.go

```diff
@@ -40,6 +40,9 @@ type Storage interface {
 	// label matchers. At least one label matcher must be specified that does not
 	// match the empty string.
 	MetricsForLabelMatchers(matchers ...*metric.LabelMatcher) map[clientmodel.Fingerprint]clientmodel.COWMetric
+	// LastSamplePairForFingerprint returns the last sample pair for the provided fingerprint.
+	// If the respective time series is evicted, nil is returned.
```

Comment:
Nit: The correct term is "archived". (We _evict_ chunks but _archive_ time series (which can happen after all chunks of the time series have been evicted).)